File: design-system/android/DesignSystem.kt
package com.futinfo.design

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// Design System for Android (Jetpack Compose)

object DesignSystem {
    
    // Colors
    object Colors {
        val primary = Color(0xFF1E88E5)
        val primaryLight = Color(0xFF64B5F6)
        val primaryDark = Color(0xFF1565C0)
        
        object Trust {
            val official = Color(0xFF4CAF50)
            val tier1 = Color(0xFF2196F3)
            val verified = Color(0xFF9C27B0)
            val reliable = Color(0xFF00BCD4)
            val questionable = Color(0xFFFF9800)
            val unreliable = Color(0xFFF44336)
        }
        
        object Categories {
            val general = Color(0xFF757575)
            val transfer = Color(0xFFFF9800)
            val match = Color(0xFF2196F3)
            val injury = Color(0xFF9C27B0)
        }
        
        object Background {
            val primary = Color(0xFFFAFAFA)
            val secondary = Color(0xFFF5F5F5)
            val card = Color.White
            val dark = Color(0xFF121212)
        }
        
        object Text {
            val primary = Color(0xFF212121)
            val secondary = Color(0xFF757575)
            val disabled = Color(0xFFBDBDBD)
            val inverse = Color.White
        }
        
        val divider = Color(0x1F000000)
        val shadow = Color(0x1A000000)
    }
    
    // Typography
    object Typography {
        val headline = androidx.compose.ui.text.TextStyle(
            fontSize = 18.sp,
            fontWeight = FontWeight.SemiBold
        )
        
        val subheadline = androidx.compose.ui.text.TextStyle(
            fontSize = 16.sp,
            fontWeight = FontWeight.Medium
        )
        
        val body = androidx.compose.ui.text.TextStyle(
            fontSize = 16.sp,
            fontWeight = FontWeight.Normal
        )
        
        val caption = androidx.compose.ui.text.TextStyle(
            fontSize = 14.sp,
            fontWeight = FontWeight.Normal
        )
        
        val small = androidx.compose.ui.text.TextStyle(
            fontSize = 12.sp,
            fontWeight = FontWeight.Normal
        )
    }
    
    // Spacing
    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 16.dp
        val lg = 24.dp
        val xl = 32.dp
        val xxl = 48.dp
    }
    
    // Border Radius
    object BorderRadius {
        val sm = 4.dp
        val md = 8.dp
        val lg = 12.dp
        val xl = 16.dp
        val full = 50.dp
    }
}

// Common Components

@Composable
fun DSCard(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    Card(
        modifier = modifier,
        shape = RoundedCornerShape(DesignSystem.BorderRadius.lg),
        colors = CardDefaults.cardColors(
            containerColor = DesignSystem.Colors.Background.card
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 2.dp
        )
    ) {
        Column(
            modifier = Modifier.padding(DesignSystem.Spacing.md),
            content = content
        )
    }
}

@Composable
fun DSButton(
    text: String,
    onClick: () -> Unit,
    style: ButtonStyle = ButtonStyle.Primary,
    modifier: Modifier = Modifier
) {
    val colors = when (style) {
        ButtonStyle.Primary -> ButtonDefaults.buttonColors(
            containerColor = DesignSystem.Colors.primary,
            contentColor = DesignSystem.Colors.Text.inverse
        )
        ButtonStyle.Secondary -> ButtonDefaults.buttonColors(
            containerColor = DesignSystem.Colors.Background.secondary,
            contentColor = DesignSystem.Colors.Text.primary
        )
        ButtonStyle.Text -> ButtonDefaults.textButtonColors(
            contentColor = DesignSystem.Colors.primary
        )
    }
    
    when (style) {
        ButtonStyle.Text -> {
            TextButton(
                onClick = onClick,
                modifier = modifier,
                colors = colors
            ) {
                Text(text, style = DesignSystem.Typography.subheadline)
            }
        }
        else -> {
            Button(
                onClick = onClick,
                modifier = modifier,
                shape = RoundedCornerShape(DesignSystem.BorderRadius.full),
                colors = colors,
                contentPadding = PaddingValues(
                    horizontal = DesignSystem.Spacing.lg,
                    vertical = DesignSystem.Spacing.sm
                )
            ) {
                Text(text, style = DesignSystem.Typography.subheadline)
            }
        }
    }
}

@Composable
fun DSBadge(
    text: String,
    color: Color,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(DesignSystem.BorderRadius.sm),
        color = color.copy(alpha = 0.1f)
    ) {
        Text(
            text = text,
            modifier = Modifier.padding(
                horizontal = DesignSystem.Spacing.sm,
                vertical = DesignSystem.Spacing.xs
            ),
            style = DesignSystem.Typography.small,
            color = color,
            fontWeight = FontWeight.Medium
        )
    }
}

// News Specific Components

@Composable
fun NewsTrustIndicator(
    tier: String,
    trustScore: Int,
    modifier: Modifier = Modifier
) {
    val (color, label) = when (tier) {
        "official" -> DesignSystem.Colors.Trust.official to "[OFFICIAL]"
        "tier1" -> DesignSystem.Colors.Trust.tier1 to "[Tier 1]"
        "verified" -> DesignSystem.Colors.Trust.verified to "[Verified]"
        "reliable" -> DesignSystem.Colors.Trust.reliable to "[Reliable]"
        "questionable" -> DesignSystem.Colors.Trust.questionable to "[Rumour]"
        else -> DesignSystem.Colors.Trust.unreliable to "[Unverified]"
    }
    
    Row(
        modifier = modifier,
        horizontalArrangement = Arrangement.spacedBy(DesignSystem.Spacing.xs)
    ) {
        Box(
            modifier = Modifier
                .size(6.dp)
                .background(color, shape = androidx.compose.foundation.shape.CircleShape)
        )
        
        Text(
            text = label,
            style = DesignSystem.Typography.caption,
            color = color
        )
    }
}

@Composable
fun NewsCard(
    article: NewsArticle,
    onDuplicatesClick: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    DSCard(modifier = modifier.fillMaxWidth()) {
        // Header
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Row(horizontalArrangement = Arrangement.spacedBy(DesignSystem.Spacing.sm)) {
                Icon(
                    imageVector = getCategoryIcon(article.category),
                    contentDescription = null,
                    tint = getCategoryColor(article.category),
                    modifier = Modifier.size(16.dp)
                )
                
                Text(
                    text = article.sourceName,
                    style = DesignSystem.Typography.caption,
                    color = DesignSystem.Colors.Text.secondary
                )
                
                NewsTrustIndicator(
                    tier = article.sourceTier,
                    trustScore = article.trustScore
                )
            }
            
            if (article.duplicateCount > 0) {
                DSBadge(
                    text = "+${article.duplicateCount}",
                    color = DesignSystem.Colors.primary,
                    modifier = Modifier.clickable { onDuplicatesClick?.invoke() }
                )
            }
        }
        
        Spacer(modifier = Modifier.height(DesignSystem.Spacing.sm))
        
        // Title
        Text(
            text = article.title,
            style = DesignSystem.Typography.headline,
            color = DesignSystem.Colors.Text.primary,
            maxLines = 2
        )
        
        if (article.summary.isNotEmpty()) {
            Spacer(modifier = Modifier.height(DesignSystem.Spacing.xs))
            
            Text(
                text = article.summary,
                style = DesignSystem.Typography.body,
                color = DesignSystem.Colors.Text.secondary,
                maxLines = 2
            )
        }
        
        Spacer(modifier = Modifier.height(DesignSystem.Spacing.sm))
        
        Text(
            text = article.timeAgo,
            style = DesignSystem.Typography.small,
            color = DesignSystem.Colors.Text.disabled
        )
    }
}

// Utility Functions

fun getCategoryColor(category: String): Color {
    return when (category) {
        "transfer" -> DesignSystem.Colors.Categories.transfer
        "match" -> DesignSystem.Colors.Categories.match
        "injury" -> DesignSystem.Colors.Categories.injury
        else -> DesignSystem.Colors.Categories.general
    }
}

fun getCategoryIcon(category: String): ImageVector {
    return when (category) {
        "transfer" -> Icons.Default.SwapHoriz
        "match" -> Icons.Default.SportsSoccer
        "injury" -> Icons.Default.LocalHospital
        else -> Icons.Default.Article
    }
}

// Enums

enum class ButtonStyle {
    Primary,
    Secondary,
    Text
}

// Data Classes

data class NewsArticle(
    val id: String,
    val title: String,
    val summary: String,
    val sourceName: String,
    val sourceTier: String,
    val trustScore: Int,
    val url: String,
    val publishedAt: String,
    val category: String,
    val duplicateCount: Int = 0,
    val duplicateSources: List<String> = emptyList(),
    val timeAgo: String
)

File: design-system/ios/DesignSystem.swift
import SwiftUI

// MARK: - Design System for iOS

enum DesignSystem {
    
    // MARK: - Colors
    enum Colors {
        static let primary = Color(hex: "#1E88E5")
        static let primaryLight = Color(hex: "#64B5F6")
        static let primaryDark = Color(hex: "#1565C0")
        
        enum Trust {
            static let official = Color(hex: "#4CAF50")
            static let tier1 = Color(hex: "#2196F3")
            static let verified = Color(hex: "#9C27B0")
            static let reliable = Color(hex: "#00BCD4")
            static let questionable = Color(hex: "#FF9800")
            static let unreliable = Color(hex: "#F44336")
        }
        
        enum Categories {
            static let general = Color(hex: "#757575")
            static let transfer = Color(hex: "#FF9800")
            static let match = Color(hex: "#2196F3")
            static let injury = Color(hex: "#9C27B0")
        }
        
        enum Background {
            static let primary = Color(hex: "#FAFAFA")
            static let secondary = Color(hex: "#F5F5F5")
            static let card = Color.white
            static let dark = Color(hex: "#121212")
        }
        
        enum Text {
            static let primary = Color(hex: "#212121")
            static let secondary = Color(hex: "#757575")
            static let disabled = Color(hex: "#BDBDBD")
            static let inverse = Color.white
        }
        
        static let divider = Color.black.opacity(0.12)
        static let shadow = Color.black.opacity(0.1)
    }
    
    // MARK: - Typography
    enum Typography {
        static func headline() -> Font {
            .system(size: 18, weight: .semibold, design: .default)
        }
        
        static func subheadline() -> Font {
            .system(size: 16, weight: .medium, design: .default)
        }
        
        static func body() -> Font {
            .system(size: 16, weight: .regular, design: .default)
        }
        
        static func caption() -> Font {
            .system(size: 14, weight: .regular, design: .default)
        }
        
        static func small() -> Font {
            .system(size: 12, weight: .regular, design: .default)
        }
    }
    
    // MARK: - Spacing
    enum Spacing {
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 16
        static let lg: CGFloat = 24
        static let xl: CGFloat = 32
        static let xxl: CGFloat = 48
    }
    
    // MARK: - Border Radius
    enum BorderRadius {
        static let sm: CGFloat = 4
        static let md: CGFloat = 8
        static let lg: CGFloat = 12
        static let xl: CGFloat = 16
        static let full: CGFloat = 9999
    }
    
    // MARK: - Shadows
    enum Shadow {
        static func small() -> some View {
            Color.black.opacity(0.05)
                .shadow(radius: 2, x: 0, y: 1)
        }
        
        static func medium() -> some View {
            Color.black.opacity(0.1)
                .shadow(radius: 4, x: 0, y: 2)
        }
        
        static func large() -> some View {
            Color.black.opacity(0.1)
                .shadow(radius: 6, x: 0, y: 4)
        }
    }
}

// MARK: - Common Components

struct DSCard<Content: View>: View {
    let content: Content
    
    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }
    
    var body: some View {
        content
            .padding(DesignSystem.Spacing.md)
            .background(DesignSystem.Colors.Background.card)
            .cornerRadius(DesignSystem.BorderRadius.lg)
            .shadow(color: DesignSystem.Colors.shadow, radius: 2, x: 0, y: 1)
    }
}

struct DSButton: View {
    let title: String
    let style: ButtonStyle
    let action: () -> Void
    
    enum ButtonStyle {
        case primary
        case secondary
        case text
    }
    
    var body: some View {
        Button(action: action) {
            Text(title)
                .font(DesignSystem.Typography.subheadline())
                .foregroundColor(foregroundColor)
                .padding(.horizontal, DesignSystem.Spacing.lg)
                .padding(.vertical, DesignSystem.Spacing.sm)
                .background(backgroundColor)
                .cornerRadius(DesignSystem.BorderRadius.full)
        }
    }
    
    private var backgroundColor: Color {
        switch style {
        case .primary: return DesignSystem.Colors.primary
        case .secondary: return DesignSystem.Colors.Background.secondary
        case .text: return Color.clear
        }
    }
    
    private var foregroundColor: Color {
        switch style {
        case .primary: return DesignSystem.Colors.Text.inverse
        case .secondary: return DesignSystem.Colors.Text.primary
        case .text: return DesignSystem.Colors.primary
        }
    }
}

struct DSBadge: View {
    let text: String
    let color: Color
    
    var body: some View {
        Text(text)
            .font(DesignSystem.Typography.small())
            .fontWeight(.medium)
            .foregroundColor(color)
            .padding(.horizontal, DesignSystem.Spacing.sm)
            .padding(.vertical, DesignSystem.Spacing.xs)
            .background(color.opacity(0.1))
            .cornerRadius(DesignSystem.BorderRadius.sm)
    }
}

// MARK: - News Specific Components

struct NewsTrustIndicator: View {
    let tier: String
    let trustScore: Int
    
    var body: some View {
        HStack(spacing: DesignSystem.Spacing.xs) {
            Circle()
                .fill(trustColor)
                .frame(width: 6, height: 6)
            
            Text(tierLabel)
                .font(DesignSystem.Typography.caption())
                .foregroundColor(trustColor)
        }
    }
    
    private var trustColor: Color {
        switch tier {
        case "official": return DesignSystem.Colors.Trust.official
        case "tier1": return DesignSystem.Colors.Trust.tier1
        case "verified": return DesignSystem.Colors.Trust.verified
        case "reliable": return DesignSystem.Colors.Trust.reliable
        case "questionable": return DesignSystem.Colors.Trust.questionable
        default: return DesignSystem.Colors.Trust.unreliable
        }
    }
    
    private var tierLabel: String {
        switch tier {
        case "official": return "[OFFICIAL]"
        case "tier1": return "[Tier 1]"
        case "verified": return "[Verified]"
        case "reliable": return "[Reliable]"
        case "questionable": return "[Rumour]"
        default: return "[Unverified]"
        }
    }
}

// MARK: - Extensions

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3:
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6:
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8:
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}